File: 14. Longest Common Prefix/14.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        pre = strs[0]
        
        for i in range(1, len(strs)):
            while pre not in strs[i][:len(pre)]:
                #print(pre, strs[i][:len(pre)])
                pre = pre[:-1]
                if pre == "":
                    return pre
        return pre

File: 14. Longest Common Prefix/14.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        string pre = strs[0];
        for (int i = 1; i < strs.Length; i++) 
        {
            while (!strs[i].StartsWith(pre) && pre.Length > 0)
            {

                pre = pre.Substring(0, pre.Length - 1);
                
                if (pre.Length == 0) 
                {
                    return "";
                }
            }
        }
        return pre;
    }
}